File: pub & sub/task1_ws/src/string_pkg/src/pub_string.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <sstream>

using namespace ros;

int main(int argc, char **argv)
{
    // node
    init(argc, argv, "talker", init_options::AnonymousName);
    NodeHandle nh;

    // publisher
    Rate rate(1);
    Publisher pub = nh.advertise<std_msgs::String>("string_topic", 1);

    // message
    std_msgs::String msg;
    std::stringstream ss;
    ss << "I am Learning Robotics and ROS";
    msg.data = ss.str();

    // publishing
    while (ok())
    {
        pub.publish(msg);
        spinOnce();
        rate.sleep();
    }

    return 0;
}

File: pub_sub/pub_sub_ws/src/string_pkg/scripts/pub_string.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import String


def talker():
    # create node
    rospy.init_node("talker", anonymous=True)
    # create publisher object
    pub = rospy.Publisher("string_topic", String, queue_size=1)
    # create rate object
    rate = rospy.Rate(10)

    # define msg and sending loop
    msg = "I am Learning Robotics and ROS"
    while not rospy.is_shutdown():
        pub.publish(msg)
        rate.sleep() # pause loop ro reach desired rate frequency



if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInternalException:
        pass